main.nf

```diff
@@ -198,15 +206,35 @@ Channel
         analysis_id = list[2]
         [ chip_sample_id, ctrl_sample_id, analysis_id ]
     }
-    .into{ macs_para; saturation_para }
+    .into{ vali_para; macs_para; saturation_para }
+
+// Validate all samples in macs config file
+def config_samples = []
+for (line in vali_para){
+    if (line.getClass().toString() != "class groovyx.gpars.dataflow.operator.PoisonPill") {
+        config_samples.add(line[0])
+        config_samples.add(line[1])
+    }
+}
+config_samples.removeAll{ it == '' }
+config_samples.unique(false)
 
-/*
- * Reference to use for MACS, ngs.plot.r and annotation
- */
-def REF_macs = false
-def REF_ngsplot = false
-if (params.genome == 'GRCh37'){ REF_macs = 'hs'; REF_ngsplot = 'hg19' }
-else if (params.genome == 'GRCm38'){ REF_macs = 'mm'; REF_ngsplot = 'mm10' }
+def fastq_samples = []
+for (sample in raw_reads_configvali){
+    if (sample.getClass().toString() != "class groovyx.gpars.dataflow.operator.PoisonPill") {
+        fastq_samples.add(sample[0].toString() - ~/(.R)?(_R)?(.R1)?(_1)?(_R1)?(_trimmed)?(_val_1)?(\.fq)?(\.fastq)?(\.gz)?$/)
+    }
+}
+
+def missing_samples = config_samples - config_samples.intersect(fastq_samples)
+if(!missing_samples.isEmpty()){
+    exit 1, "No FastQ file found for sample in MACS config: ${missing_samples}"
+}
+
+def dropped_samples = fastq_samples - fastq_samples.intersect(config_samples)
+if(!dropped_samples.isEmpty()){
+    log.warn "Sample ${dropped_samples} not included in MACS config"
```

Comment:
Maybe `exit 1` here too? It seems like a pretty major problem to not have this included? Or is there a run scenario where this could happen?

I'm worried that a single warn statement like this would be easy to miss.